SharpPusher/Services/PushServices/Blockchair.cs

```diff
@@ -1,71 +1,145 @@
+        {
+            BTC,
+            BCH,
+            DOGE,
+            LTC,
+            XMR,
+            TBTC,
+            BSV,
+            ZEC,
+            XRP,
+            tETH,
+            ETH,
+            EOS,
+            XTZ,
+            XIN,
+            ADA,
+            XLM,
+            GRS,
+            DASH,
+            ABC
+        }
+
+        private readonly Chain chain;
+
+        public override string ApiName => "Blockchair";
+
+        public override async Task<Response<string>> PushTx(string txHex)
+        {
+            Response<string> resp = new Response<string>();
+
+            using (HttpClient client = new HttpClient())
+            {
+                try
+                {
+                    string chainName = "";
+                    if(chain == Chain.BTC)
+                    {
+                        chainName = "bitcoin";
+                    }else if(chain == Chain.TBTC)
+                    {
+                        chainName = "bitcoin/testnet";
+                    }else if(chain == Chain.BCH)
+                    {
+                        chainName = "bitcoin-cash";
+                    }else if(chain == Chain.DOGE)
+                    {
+                        chainName = "dogecoin";
+                    }else if(chain == Chain.LTC)
+                    {
+                        chainName = "litecoin";
+                    }else if(chain == Chain.XMR)
+                    {
+                        chainName = "monero";
+                    }else if(chain == Chain.ADA)
+                    {
+                        chainName = "cardano";
+                    }else if(chain == Chain.BSV)
+                    {
+                        chainName = "bitcoin-sv";
+                    }else if(chain == Chain.EOS)
+                    {
+                        chainName = "eos";
+                    }else if(chain == Chain.ETH)
+                    {
+                        chainName = "ethereum";
+                    }else if(chain == Chain.tETH)
+                    {
+                        chainName = "ethereum/testnet";
+                    }else if(chain == Chain.XIN)
+                    {
+                        chainName = "mixin";
+                    }else if(chain == Chain.XLM)
+                    {
+                        chainName = "stellar";
+                    }else if(chain == Chain.XRP)
+                    {
+                        chainName = "ripple";
+                    }else if(chain == Chain.XTZ)
+                    {
+                        chainName = "tezos";
+                    }else if (chain == Chain.DASH)
+                    {
+                        chainName = "dash";
+                    }else if (chain == Chain.GRS)
+                    {
+                        chainName = "groestlcoin";
+                    }else if (chain == Chain.ABC)
+                    {
+                        chainName = "bitcoin-abc";
+                    }

```

Comment:
Please use a switch with the new and shorter [switch expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression) here.  

    string chainName = chain switch
    {
        Chain.Foo => "Foo",
        ....
        _ => Throw new ArgumentException("Undefined Chain")
    }

Reply:
Thanks! I didn't know about that. Lines 52-74 are now <a href="https://github.com/AngeloMetal/SharpPusher/blob/master/SharpPusher/Services/PushServices/Blockchair.cs#L52-L74">edited</a>.